var/spack/repos/builtin/packages/gaudi/package.py

```diff
@@ -42,13 +43,15 @@ class Gaudi(CMakePackage):
 
     depends_on("cxx", type="build")  # generated
 
+    conflicts("%gcc@:10", when="@39:")
```

Comment:
```suggestion
    conflicts("%gcc@:10", when="@39:", msg="Gaudi needs a c++20 capable compiler for this version")
```
Just to give users a bit more information about what is going on.